dist/tools/pkg_version_check/version_check.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2024 Nam Nguyen
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os
+import requests
+from concurrent.futures import ThreadPoolExecutor, as_completed
+import threading
+import shutil
+
+
+def get_makefiles():
+    # Get the makefiles and read the PKG_URL and PKG_VERSION
+    file_contents = {}
+    # Get the current working directory
+    relative_path = os.path.join(os.path.dirname(__file__), '..', '..', '..', 'pkg')
+    current_dir = os.path.abspath(relative_path)
+    # Iterate through each item in the current directory
+    for item in os.listdir(current_dir):
+        # Construct the full path of the item
+        item_path = os.path.join(current_dir, item)
+        # Check if the item is a directory
+        if os.path.isdir(item_path):
+            # Construct the path of the Makefile within the directory
+            makefile_path = os.path.join(item_path, 'Makefile')
+            # Check if Makefile exists in the directory
+            if os.path.isfile(makefile_path):
+                # Open the Makefile and read its content
+                with open(makefile_path, 'r') as file:
+                    content = file.readlines()
+                # Iterate through each line to find PKG_VERSION and PKG_URL
+                pkg_version = None
+                pkg_url = None
+                for line in content:
+                    # Check if line contains PKG_URL
+                    if line.strip().startswith("PKG_URL"):
+                        # Extract the value of PKG_URL
+                        pkg_url = line.strip().split("=")[1]
+                        # Process space and comments
+                        if pkg_url[0] == " ":
+                            pkg_url = pkg_url[1:]
+                        pkg_url = pkg_url.split(" ")[0]
```

Comment:
```suggestion
                        pkg_url = pkg_url.strip()
```
shouldn't this do the trick already? same below

Reply:
I used split because it splits the `PKG_URL` line into an array. This makes it easier to remove the inline comments as I could just select the 0th element of the array to get the `PKG_URL` only.